src/rocksdb_options.rs

```diff
@@ -379,6 +381,16 @@ impl Options {
         }
     }
 
+    pub fn add_table_properties_collector_factory(&mut self,
+                                                  factory: Box<TablePropertiesCollectorFactory>) {
+        unsafe {
+            let context = new_table_properties_collector_factory_context(factory);
+            let context = Box::into_raw(context) as *mut c_void;
```

Comment:
context will be freed in the context destructor.
https://github.com/pingcap/rust-rocksdb/pull/75/files/088c152abd9baa6196813c8768240a5081ca3bee#diff-8531ef5a3950c5c49f24b86d76ea8a32R39